app/javascript/packages/document-capture/higher-order/with-background-encrypted-upload.jsx

```diff
@@ -46,58 +52,66 @@ export async function encrypt(key, iv, value) {
   );
 }
 
-const withBackgroundEncryptedUpload = (Component) => ({ onChange, ...props }) => {
-  const { backgroundUploadURLs, backgroundUploadEncryptKey } = useContext(UploadContext);
-  const { addPageAction } = useContext(AnalyticsContext);
-
+const withBackgroundEncryptedUpload = (Component) =>
   /**
-   * @param {Record<string, string|Blob|null|undefined>} nextValues Next values.
+   * @param {Pick<FormStepComponentProps<Record<string,any>>, 'onChange'|'onError'>} props
    */
-  function onChangeWithBackgroundEncryptedUpload(nextValues) {
-    const nextValuesWithUpload = {};
-    for (const [key, value] of Object.entries(nextValues)) {
-      nextValuesWithUpload[key] = value;
-      const url = backgroundUploadURLs[key];
-      if (url && value) {
-        const iv = window.crypto.getRandomValues(new Uint8Array(12));
-        nextValuesWithUpload[`${key}_image_iv`] = window.btoa(String.fromCharCode(...iv));
-        nextValuesWithUpload[`${key}_image_url`] = encrypt(
-          /** @type {CryptoKey} */ (backgroundUploadEncryptKey),
-          iv,
-          value,
-        )
-          .then((encryptedValue) =>
-            window.fetch(url, {
-              method: 'PUT',
-              body: encryptedValue,
-              headers: { 'Content-Type': 'application/octet-stream' },
-            }),
+  ({ onChange, onError, ...props }) => {
```

Comment:
Recommend ["Hide Whitespace Changes" setting](https://user-images.githubusercontent.com/1779930/109327839-c4285500-7826-11eb-960d-d0fe88e69b25.png). It's just indentation.

Reply:
Will our TS checks error if we construct this component and do not provide an `onError`?

Reply:
> Will our TS checks error if we construct this component and do not provide an `onError`?

The individual components should define the type themselves ([example](https://github.com/18F/identity-idp/blob/f70fa9894dc6fec44228b00b9c3e5827a61816f8/app/javascript/packages/document-capture/components/selfie-step.jsx#L23-L25)). Adding the type here was more a convenience to double-check the usage of `onChange` and `onError` within this implementation.